tests/Data/DataResponseTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Tests\Data;
+
+use Nyholm\Psr7\Factory\Psr17Factory;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ResponseInterface;
+use Yiisoft\Http\Status;
+use Yiisoft\Yii\Web\Data\Formatter\JsonDataResponseFormatter;
+use Yiisoft\Yii\Web\Data\DataResponse;
+
+class DataResponseTest extends TestCase
+{
+    public function testCreateResponse(): void
+    {
+        $factory = new Psr17Factory();
+        $dataResponse = new DataResponse('test', Status::OK, '', $factory);
+        $dataResponse = $dataResponse->withHeader('Content-Type', 'application/json');
+        $dataResponse->getBody()->rewind();
+
+        $this->assertInstanceOf(ResponseInterface::class, $dataResponse);
+        $this->assertSame(['application/json'], $dataResponse->getResponse()->getHeader('Content-Type'));
+        $this->assertSame(['application/json'], $dataResponse->getHeader('Content-Type'));
+        $this->assertSame($dataResponse->getResponse()->getBody(), $dataResponse->getBody());
+        $this->assertSame('test', $dataResponse->getBody()->getContents());
+    }
+
+    public function testChangeResponseData(): void
+    {
+        $factory = new Psr17Factory();
```

Comment:
We need to test the `DataResponse`. We no need to test any factories. 
Use mock for `DataResponse` instead of directly creating

Reply:
It's OK to use real objects if they're lightweight enough but I'll accept a pull request fixing it.